example-app/src/main/java/backtraceio/backtraceio/MainActivity.java

```diff
@@ -33,9 +43,37 @@ protected void onCreate(Bundle savedInstanceState) {
         settings.setRetryOrder(RetryOrder.Queue);
 
         BacktraceDatabase database = new BacktraceDatabase(context, settings);
-        BacktraceClient backtraceClient = new BacktraceClient(context, credentials, database);
+        backtraceClient = new BacktraceClient(context, credentials, database);
 
         BacktraceExceptionHandler.enable(backtraceClient);
         backtraceClient.send("test");
+
+        // Enable handling of native crashes
+        database.setupNativeIntegration(backtraceClient, credentials);
+
+        // Enable ANR detection with a timeout of 3000 milliseconds
+        backtraceClient.enableAnr(3000);
+    }
+
+    public native void cppCrash();
+
+    public void handledException(View view) {
+        try {
+            throw new Exception("Handle this exception and report to Backtrace");
+        } catch (Exception e) {
+            backtraceClient.send(new BacktraceReport(e));
+        }
+    }
+
+    public void unhandledException(View view) {
+        throw new NullPointerException("My null ptr exception");
+    }
+
+    public void nativeCrash(View view) {
+        cppCrash();
+    }
+
+    public void anr(View view) throws InterruptedException {
+        Thread.sleep(6000);
```

Comment:
I think if user will change anr value for testing purposes and then fire "ANR" button, then he won't see anr in worst case. Instead I propose to expose ANR timeout to private const variable of a class and then in this method sleep for _anrTimeout + 1sec time.

Reply:
I had to increase the slack from your proposed 1sec to 2sec to force the ANR